syntax = "proto3";

message RegisterImplant {
    string ImplantID = 1;
    string ComputerName = 2;
    string PublicKey = 3;
    string Username = 4;
    string Password = 5;
}


message TaskRequest {
  int32 TaskID = 1;
  int32 Opcode = 2;
  string Args = 3;
}

message TaskResponse {
  int32 TaskID = 1;
  string ImplantID = 2; //implant ID
  bytes Response = 3;
}

message ImplantCheckin {
  string ImplantID = 1;
  TaskResponse Resp = 2;
}
